packages/build/src/test/type_test.ts

```diff
@@ -88,87 +88,123 @@ test("anyOf", () => {
-        properties: {
-          bar: {
-            type: "string",
+  test("object with nested object", () => {
+    const obj3 = object({ foo: object({ bar: "string" }) });
+    // $ExpectType { foo: { bar: string; }; }
+    type t3 = ConvertBreadboardType<typeof obj3>;
+    assert.deepEqual(toJSONSchema(obj3), {
+      type: "object",
+      properties: {
+        foo: {
+          type: "object",
+          properties: {
+            bar: {
+              type: "string",
+            },
           },
+          required: ["bar"],
         },
-        required: ["bar"],
       },
-    },
-    required: ["foo"],
-  });
-
-  const obj4 = object({ foo: anyOf("string", "number") });
-  // $ExpectType { foo: string | number; }
-  type t4 = ConvertBreadboardType<typeof obj4>;
-  assert.deepEqual(toJSONSchema(obj4), {
-    type: "object",
-    properties: {
-      foo: { anyOf: [{ type: "string" }, { type: "number" }] },
-    },
-    required: ["foo"],
-  });
-
-  const obj5 = object({ foo: anyOf("string", object({ bar: "string" })) });
-  // $ExpectType { foo: string | { bar: string; }; }
-  type t5 = ConvertBreadboardType<typeof obj5>;
-  assert.deepEqual(toJSONSchema(obj5), {
-    type: "object",
-    properties: {
-      foo: {
-        anyOf: [
-          { type: "string" },
-          {
-            type: "object",
-            properties: { bar: { type: "string" } },
-            required: ["bar"],
-          },
-        ],
+      required: ["foo"],
+    });
+  });
+
+  test("object with anyOf type", () => {
+    const obj4 = object({ foo: anyOf("string", "number") });
+    // $ExpectType { foo: string | number; }
+    type t4 = ConvertBreadboardType<typeof obj4>;
+    assert.deepEqual(toJSONSchema(obj4), {
+      type: "object",
+      properties: {
+        foo: { anyOf: [{ type: "string" }, { type: "number" }] },
       },
-    },
-    required: ["foo"],
+      required: ["foo"],
+    });
+  });
+
+  test("object with anyOf type including object", () => {
+    const obj5 = object({ foo: anyOf("string", object({ bar: "string" })) });
+    // $ExpectType { foo: string | { bar: string; }; }
+    type t5 = ConvertBreadboardType<typeof obj5>;
+    assert.deepEqual(toJSONSchema(obj5), {
+      type: "object",
+      properties: {
+        foo: {
+          anyOf: [
+            { type: "string" },
+            {
+              type: "object",
+              properties: { bar: { type: "string" } },
+              required: ["bar"],
+            },
+          ],
+        },
+      },
+      required: ["foo"],
+    });
+  });
+
+  test("object with unknown property", () => {
+    const obj = object({ foo: "unknown" });
+    // $ExpectType { foo: unknown; }
+    type objType = ConvertBreadboardType<typeof obj>;
+    assert.deepEqual(toJSONSchema(obj), {
+      type: "object",
+      properties: {
+        foo: {},
+      },
+      required: ["foo"],
+    });
   });
 
+  test("object no known properties", () => {
+    const obj = object({});
```

Comment:
my first instinct was to try and write it like this. I'm not sure if this would better reflect the desired behaviour and be more intuitive for users.
```suggestion
    const obj = object("unknown");
```

Reply:
I think I like it how it is, it's more consistent (where the argument to object is an object describing the properties; if it can also be a basic type in some cases, it's not quite so obvious what that means IMO).